File: 30 Days of Code/C/Day20_Sorting.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

void bubbleSort(int a[], int n)
{
    int swaps = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - 1; j++) {
            if (a[j] > a[j + 1]) {
                int temp;
                temp = a[j+1];
                a[j+1] = a[j];
                a[j] = temp;
                swaps++;
            }
        }
    }
    
    printf("Array is sorted in %d swaps.\n", swaps);
    printf("First Element: %d\n", a[0]);
    printf("Last Element: %d\n", a[n-1]);
}

int main()
{
    int n;
    scanf("%d", &n);
    
    int *arr=(int*)malloc(n*sizeof(int));
    
    for(int i=0; i<n; i++){
        scanf("%d", &arr[i]);
    }
    bubbleSort(arr, n);
    
    return 0;
}

File: 30 Days of Code/Java/Day20_Sorting.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;



public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        // Write your code here
        int []arr = new int[a.size()];
        for(int i=0; i<arr.length; i++)
            arr[i] = a.get(i);
        
        int swaps = 0;
        for(int i=0; i<arr.length; i++){
            for(int j=0; j<arr.length-1; j++){
                if(arr[j] > arr[j+1]){
                    int temp  =arr[j+1];
                    arr[j+1] = arr[j];
                    arr[j] = temp;
                    swaps = swaps + 1;
                }
            }
        }
        
        System.out.println("Array is sorted in " + swaps + " swaps.");
        System.out.println("First Element: "+ arr[0]);
        System.out.println("Last Element: "+arr[n-1]);

        bufferedReader.close();
    }
}
